compiler/rustc_resolve/src/check_unused.rs

```diff
@@ -320,15 +320,19 @@ impl Resolver<'_> {
             );
             let test_module_span = match module_to_string(parent_module) {
                 Some(module)
-                    if module == "test"
+                    // If we are in the `--test` mode, suppress a help that adds the `#[cfg(test)]`
+                    // attribute; however, if not, suggest adding the attribute. There is no way to
+                    // retrieve attributes here because we do not have a `TyCtxt` yet.
+                    if !visitor.r.session.opts.test
```

Comment:
Could you move this check outwards, so we don't needlessly compute `nearest_non_block_module`?
r=me once this is done

Reply:
@cjgillot 

Thank you! I've fixed it.

I don't think I have sufficient permission to approve this PR, so could you please review this PR again?